.github/workflows/build.yml

```diff
@@ -14,6 +14,7 @@ permissions:
   contents: read
 
 env:
+  RUSTFLAGS: -Dwarnings
```

Comment:
@Freax13 @josephlr What do you think about failing the CI on warnings?

Reply:
Similar changes have been done in https://github.com/rust-osdev/x86_64/pull/475.

Another option would be to add additional jobs that fail on warnings but are allowed to fail.

Reply:
I think it's a good idea to do that. My only concern is that we're using nightly and CI might break more often than it would on stable. I guess we'll have to see how often this occurs in practice. If it's too often we can always disable it again.

According to [the docs](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/notifications-for-workflow-runs), @phil-opp should be the one getting notifications whenever our scheduled workflow fails because he was the last person to touch the cron syntax. AFAICT there isn't a way to send notifications to more than one person. If you don't want to deal with those notifications, I'd also be happy to take over.

Reply:
I would like to deny warnings in all configurations for now. If it becomes an issue we can either:
  - Disable it for `nightly` toolchains
  - Disable it for "weird" configurations

Reply:
Ok, sounds good to me!

I'm fine with receiving the notifications. I'll open an issue when I notice a build error.